#Use external CGAL and METIS (via CGAL_ROOT, BOOST_ROOT and METIS_ROOT in easyconfig)
#B. Hajgato Nov 16 2015
 # -----------------------------------------------------------------------------
--- OpenFOAM-3.0.0/etc/config/CGAL.csh	2015-11-16 13:54:59.897564794 +0100
+++ OpenFOAM-3.0.0.orig/etc/config/CGAL.csh	2015-11-16 13:46:45.131988362 +0100
@@ -29,26 +29,36 @@
 #     Sourced from OpenFOAM-<VERSION>/etc/cshrc
 ##------------------------------------------------------------------------------
 
-set boost_version=boost-system
-set cgal_version=CGAL-4.7
+if ( ${?EBROOTCGAL} != 0 && ${?EBROOTBOOST} != 0) then
+   setenv BOOST_ARCH_PATH $EBROOTBOOST
+   setenv CGAL_ARCH_PATH $EBROOTCGAL
+
+elif ( ${?CGAL_ROOT} != 0 && ${?BOOST_ROOT}) then
+   setenv BOOST_ARCH_PATH $BOOST_ROOT
+   setenv CGAL_ARCH_PATH $CGAL_ROOT
+
+else
+    set boost_version=boost-system
+    set cgal_version=CGAL-4.7
+
+    setenv BOOST_ARCH_PATH $WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$boost_version
+    setenv CGAL_ARCH_PATH $WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$cgal_version
+
+    if ($?FOAM_VERBOSE && $?prompt) then
+        echo "Using CGAL and boost"
+        echo "    $cgal_version at $CGAL_ARCH_PATH"
+        echo "    $boost_version at $BOOST_ARCH_PATH"
+    endif
+
+    if ( -d "$CGAL_ARCH_PATH" ) then
+        _foamAddLib $CGAL_ARCH_PATH/lib
+    endif
+
+    if ( -d "$BOOST_ARCH_PATH" ) then
+        _foamAddLib $BOOST_ARCH_PATH/lib
+    endif
 
-setenv BOOST_ARCH_PATH $WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$boost_version
-setenv CGAL_ARCH_PATH $WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$cgal_version
-
-if ($?FOAM_VERBOSE && $?prompt) then
-    echo "Using CGAL and boost"
-    echo "    $cgal_version at $CGAL_ARCH_PATH"
-    echo "    $boost_version at $BOOST_ARCH_PATH"
-endif
-
-if ( -d "$CGAL_ARCH_PATH" ) then
-    _foamAddLib $CGAL_ARCH_PATH/lib
+    unset boost_version cgal_version
 endif
 
-if ( -d "$BOOST_ARCH_PATH" ) then
-    _foamAddLib $BOOST_ARCH_PATH/lib
-endif
-
-unset boost_version cgal_version
-
 # -----------------------------------------------------------------------------
--- OpenFOAM-3.0.0/etc/config/CGAL.sh	2015-11-16 13:54:59.897564794 +0100
+++ OpenFOAM-3.0.0.orig/etc/config/CGAL.sh	2015-11-03 11:30:23.000000000 +0100
@@ -29,29 +29,39 @@
 #     Sourced from OpenFOAM-<VERSION>/etc/bashrc
 #------------------------------------------------------------------------------

-boost_version=boost-system
-cgal_version=CGAL-4.7
+if [ -n "${EBROOTCGAL}" ] && [ -n "${EBROOTBOOST}" ]; then
+    export CGAL_ARCH_PATH="${EBROOTCGAL}"
+    export BOOST_ARCH_PATH="${EBROOTBOOST}"
+
+elif [ -n "${CGAL_ROOT}" ] && [ -n "${BOOST_ROOT}" ]; then
+    export CGAL_ARCH_PATH="${BOOST_ROOT}"
+    export BOOST_ARCH_PATH="${CGAL_ROOT}"
+
+else
+    boost_version=boost-system
+    cgal_version=CGAL-4.7
+
+    export BOOST_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$boost_version
+    export CGAL_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$cgal_version
+
+    if [ "$FOAM_VERBOSE" -a "$PS1" ]
+    then
+        echo "Using CGAL and boost" 1>&2
+        echo "    $cgal_version at $CGAL_ARCH_PATH" 1>&2
+        echo "    $boost_version at $BOOST_ARCH_PATH" 1>&2
+    fi
+
+    if [ -d "$CGAL_ARCH_PATH" ]
+    then
+        _foamAddLib $CGAL_ARCH_PATH/lib
+    fi
+
+    if [ -d "$BOOST_ARCH_PATH" ]
+    then
+        _foamAddLib $BOOST_ARCH_PATH/lib
+    fi
 
-export BOOST_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$boost_version
-export CGAL_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$cgal_version
-
-if [ "$FOAM_VERBOSE" -a "$PS1" ]
-then
-    echo "Using CGAL and boost" 1>&2
-    echo "    $cgal_version at $CGAL_ARCH_PATH" 1>&2
-    echo "    $boost_version at $BOOST_ARCH_PATH" 1>&2
-fi
-
-if [ -d "$CGAL_ARCH_PATH" ]
-then
-    _foamAddLib $CGAL_ARCH_PATH/lib
+    unset boost_version cgal_version
 fi
 
-if [ -d "$BOOST_ARCH_PATH" ]
-then
-    _foamAddLib $BOOST_ARCH_PATH/lib
-fi
-
-unset boost_version cgal_version
-
 # -----------------------------------------------------------------------------
--- OpenFOAM-3.0.0/etc/config/metis.sh	2015-11-16 13:55:03.887561039 +0100
+++ OpenFOAM-3.0.0.orig/etc/config/metis.sh	2015-11-16 13:46:45.132988361 +0100
@@ -33,7 +33,15 @@
 #     during the wmake process
 #------------------------------------------------------------------------------
 
-export METIS_VERSION=metis-5.1.0
-export METIS_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$METIS_VERSION
+if [ -n "${EBROOTMETIS}" ] ; then
+    export METIS_ARCH_PATH=$EBROOTMETIS
+
+elif [ -n "${METIS_ROOT}" ] ; then
+    export METIS_ARCH_PATH=$METIS_ROOT
+
+else
+    export METIS_VERSION=metis-5.1.0
+    export METIS_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$METIS_VERSION
+fi
 
 # -----------------------------------------------------------------------------
